### PR TITLE
Add command for transferring ownership & key backup

### PR DESCRIPTION
**Pull request by joehand**
Adds a `dat keys` command a two subcommands: `dat keys import` and `dat keys export`.

* `dat keys` - prints out dat key and discovery key
* `dat keys export` - prints out secret key (does not remove ownership from existing dat)
* `dat keys import` - prompts user for secret key to make a dat writable.

These are still somewhat advanced key features but will *at least* make it possible to backup or transfer ownership of a dat.

(#827)

### CODE REVIEW

**Comment by ralphtheninja** [comment]
LGTM! Really nicely implemented.

File: src/commands/keys.js [hunk 1]
```diff
@@ -102,11 +88,17 @@ function run (dat, opts) {
...
-        process.exit(1)
-      }
+      if (err) return exit(err)
       console.log('Successful import. Dat is now writable.')
+      exit()
     }
   }
 }
+
+function exit (err) {
```
Comment (ralphtheninja) [inline src/commands/keys.js]: :+1:

**Comment by joehand** [comment]
FYI: Waiting to fix tests before merge. Thinking that https://github.com/mafintosh/hypercore/issues/109 is causing the test errors but gonna debug a bit more when mafintosh gets into town.

**Comment by austinfrey** [comment]
this is documented in the README but I don't believe it's listed in the initial `dat --help` printout from the command line. `dat keys --help` will give details though. this could very well be intentional, but thought I'd mention it.

**Comment by joehand** [comment]
Thanks, yes it is a bit intentional. Importing/Exporting via `dat keys` still has the potential to corrupt a dat with incorrect use. We'll leave it for more advanced use cases until the multiwriter lands to avoid that.

**Pull request merged**